File: frontend/packages/e2e-testing/src/__generated__/types/UpdateStackServiceVolumeCommand.ts
import type { StackServiceVolumeType } from './StackServiceVolumeType'

export type UpdateStackServiceVolumeCommand = {
  /**
   * @type string, uuid
   */
  id: string
  /**
   * @type string, uuid
   */
  stackId: string
  /**
   * @type string, uuid
   */
  stackServiceId: string
  /**
   * @type string
   */
  name: string
  /**
   * @type integer
   */
  type: StackServiceVolumeType
  /**
   * @type string
   */
  source: string | null
  /**
   * @type string
   */
  target: string
}

File: backend/src/CloudCrafter.Core/Commands/Stacks/Volumes/UpdateStackServiceVolumeCommand.cs
using AutoMapper;
using CloudCrafter.Core.Common.Security;
using CloudCrafter.Core.Interfaces.Domain.Stacks;
using CloudCrafter.Domain.Entities;
using MediatR;

namespace CloudCrafter.Core.Commands.Stacks.Volumes;

[Authorize]
public record UpdateStackServiceVolumeCommand : CreateStackServiceVolumeCommand
{
    public Guid Id { get; set; }
}

public class UpdateStackServiceVolumeCommandHandler(
    IStackServiceVolumesService stackServiceVolumesService,
    IMapper mapper
) : IRequestHandler<UpdateStackServiceVolumeCommand, Guid>
{
    public async Task<Guid> Handle(
        UpdateStackServiceVolumeCommand request,
        CancellationToken cancellationToken
    )
    {
        var type = mapper.Map<StackServiceVolumeType>(request.Type);
        var id = await stackServiceVolumesService.CreateOrUpdateStackServiceVolume(
            request.StackId,
            request.StackServiceId,
            request.Id,
            request.Name,
            type,
            request.Source,
            request.Target
        );

        return id;
    }
}
